### PR TITLE
V3.8.1 pipeline

### PR DESCRIPTION
**Pull request by star-e**
update custom pipeline implementation
fixed auto-test compile error

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by star-e** [comment]
@cocos-robot run test cases

File: native/cocos/renderer/pipeline/custom/FGDispatcherTypes.h [hunk 1]
```diff
@@ -160,6 +148,21 @@ struct BarrierNode {
     std::vector<Barrier> rearBarriers;
 };
 
+struct SliceNode {
+    bool full{false};
+    std::vector<uint32_t> mips;
```
Comment (minggo) [inline native/cocos/renderer/pipeline/custom/FGDispatcherTypes.h]: It is better to use ccstd::vector<>.

File: native/cocos/renderer/pipeline/custom/FrameGraphDispatcher.cpp [hunk 2]
```diff
@@ -122,22 +122,70 @@ ResourceGraph::vertex_descriptor FrameGraphDispatcher::realResourceID(const ccst
     return resourceAccessGraph.resourceIndex.at(name);
 }
 
+[[nodiscard]] ccstd::pmr::string concatResName(
```
Comment (minggo) [inline native/cocos/renderer/pipeline/custom/FrameGraphDispatcher.cpp:125]: I am not sure if `[[nodiscard]]` works on iOS11.

Comment (star-e) [reply native/cocos/renderer/pipeline/custom/FrameGraphDispatcher.cpp:125]: `[[nodiscard]]` has been used for more than one year. I think it works.

File: native/cocos/renderer/pipeline/custom/FGDispatcherTypes.h (hunk 1 above)
Comment (star-e) [reply native/cocos/renderer/pipeline/custom/FGDispatcherTypes.h]: renamed

**Review by minggo — APPROVED** [review]

**Pull request merged**